tests/infrastructure/node-labeller.go

```diff
@@ -0,0 +1,574 @@
+			Expect(err).ToNot(HaveOccurred())
+			Expect(val).To(BeNumerically(">", 0))
+		})
+	})
+
+	Context("advanced labelling", func() {
+		var originalKubeVirt *v1.KubeVirt
+
+		BeforeEach(func() {
+			originalKubeVirt = util.GetCurrentKv(virtClient)
+		})
+
+		AfterEach(func() {
+			tests.UpdateKubeVirtConfigValueAndWait(originalKubeVirt.Spec.Configuration)
+		})
+
+		It("[test_id:6249] should update node with new cpu model label set", func() {
+			obsoleteModel := ""
+			node := nodesWithKVM[0]
+
+			kvConfig := originalKubeVirt.Spec.Configuration.DeepCopy()
+			kvConfig.ObsoleteCPUModels = make(map[string]bool)
+
+			for key := range node.Labels {
+				if strings.Contains(key, v1.CPUModelLabel) {
+					obsoleteModel = strings.TrimPrefix(key, v1.CPUModelLabel)
+					kvConfig.ObsoleteCPUModels[obsoleteModel] = true
+					break
+				}
+			}
+
+			tests.UpdateKubeVirtConfigValueAndWait(*kvConfig)
+
+			labelKeyExpectedToBeMissing := v1.CPUModelLabel + obsoleteModel
+			expectNodeLabels(node.Name, func(m map[string]string) (valid bool, errorMsg string) {
+				_, exists := m[labelKeyExpectedToBeMissing]
+				return !exists, fmt.Sprintf("node %s is expected to not have label key %s", node.Name, labelKeyExpectedToBeMissing)
+			})
+		})
+
+		It("[test_id:6250] should update node with new cpu model vendor label", func() {
+			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			for _, node := range nodes.Items {
+				for key := range node.Labels {
+					if strings.HasPrefix(key, v1.CPUModelVendorLabel) {
+						return
+					}
+				}
+			}
+
+			Fail("No node contains label " + v1.CPUModelVendorLabel)
+		})
+
+		It("[test_id:6252] should remove all cpu model labels (all cpu model are in obsolete list)", func() {
+			node := nodesWithKVM[0]
+
+			obsoleteModels := nodelabellerutil.DefaultObsoleteCPUModels
+
+			for key := range node.Labels {
+				if strings.Contains(key, v1.CPUModelLabel) {
+					obsoleteModels[strings.TrimPrefix(key, v1.CPUModelLabel)] = true
+				}
+				if strings.Contains(key, v1.SupportedHostModelMigrationCPU) {
+					obsoleteModels[strings.TrimPrefix(key, v1.SupportedHostModelMigrationCPU)] = true
+				}
+			}
+
+			kvConfig := originalKubeVirt.Spec.Configuration.DeepCopy()
+			kvConfig.ObsoleteCPUModels = obsoleteModels
+			tests.UpdateKubeVirtConfigValueAndWait(*kvConfig)
+
+			expectNodeLabels(node.Name, func(m map[string]string) (valid bool, errorMsg string) {
+				found := false
+				label := ""
+				for key := range m {
+					if strings.Contains(key, v1.CPUModelLabel) || strings.Contains(key, v1.SupportedHostModelMigrationCPU) {
+						found = true
+						label = key
+						break
+					}
+				}
+
+				return !found, fmt.Sprintf("node %s should not contain any cpu model label, but contains %s", node.Name, label)
+			})
+		})
+	})
+
+	Context("[Serial]node with obsolete host-model cpuModel", Serial, func() {
+
+		expectSerialRun := func() {
```

Comment:
The helper functions here seem to be new.
Either changes occurred in parallel and the rebase missed them, or this needs exlaplanation and a separate commit.